app/services/groups/update_service.rb

```diff
@@ -32,7 +32,7 @@ class Groups::UpdateService < ::BaseServices::Update
   protected
 
   def persist(call)
-    removed_users = call.result.group_users.select(&:marked_for_destruction?).map(&:user)
```

Comment:
This line without the `.compact` could produce `nil` values in the resulting array which would then cause NPEs further down the line.